src/main/java/dev/dbos/transact/execution/DBOSExecutor.java

```diff
@@ -502,30 +537,59 @@ public <T> WorkflowHandle<T> submitWorkflow(String workflowName, String targetCl
         return new WorkflowHandleFuture<T>(workflowId, future, systemDatabase);
     }
 
-    public void enqueueWorkflow(String workflowName, String targetClassName,
-            WorkflowFunctionWrapper wrapper, Object[] args, Queue queue) throws Throwable {
-
-        DBOSContext ctx = DBOSContextHolder.get();
-        ctx.setDbos(dbos);
-        String wfid = ctx.getWorkflowId();
+    // TODO: add priority + deduplicationId support
+    // (https://github.com/dbos-inc/dbos-transact-java/issues/67)
+    public static String enqueueWorkflow(SystemDatabase systemDatabase, String wfid, String workflowName,
```

Comment:
@kraftp this (along with making pre/postInvokeWF static) is the change I mentioned in my [response above](https://github.com/dbos-inc/dbos-transact-java/pull/69#discussion_r2323519662)